### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kchoadley**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,4 +3,4 @@ language: generic
 
 services: docker
 
-script: docker build --rm -t aoi:test .
+script: docker build --rm -t aoi:test -f ./test/Dockerfile .
```
Comment (kchoadley) [inline .travis.yml:6]: README needs to reflect this change

File: docker-aoi-apache/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+### Create the AOI Apache web frontend image
+
+# Get base apache image from Docker Hub
+FROM httpd:2.4
+
+# Install VIM editor for configuration changes inside the machine 
+# This is not the recommended way to do it, its best to configure the file first.
+# However, it may be necessary to modify depending on the server.
+RUN apt-get update
+RUN apt-get install vim -y
```
Comment (kchoadley) [inline docker-aoi-apache/Dockerfile]: Could I get nano too?

File: docker-aoi-apache/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# AOI Apache Docker Image
+
+This utilizes version 2.4 of the Apache httpd server. The configuration file is extremely basic at present and will need to be built. As such, this has been tagged as prototype for the time being.
+
+To build locally
+
+docker build -t image-name:tag .
```
Comment (kchoadley) [inline docker-aoi-apache/README.md]: Wrap inside triple backtick for pretty formatting.
Also, use the actual command we are using, in this case 'docker build -t aoi-apache .'

File: docker-aoi-apache/README.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+Note: Replace image-name:tag with your desired nomenclature. Also note that you can pull the current prototype image from the docker hub with:
+
+docker pull os-wasabi/aoi-apache:prototype
```
Comment (kchoadley) [inline docker-aoi-apache/README.md:11]: Is prototype still the most recent tag for this docker repo?

Comment (kchoadley) [reply docker-aoi-apache/README.md:11]: If it is, I think we should consider also including a latest tag. We can version it below 1.0 if you want.

Comment (kchoadley) [reply docker-aoi-apache/README.md:11]: Wrap in triple backticks

File: docker-aoi-apache/README.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+
+Run Directions:
+To run this container you will need to provide it port information as well as a name. As below:
+
+docker run -d --name container-name -p 8080:80 aoi-apache:prototype
```
Comment (kchoadley) [inline docker-aoi-apache/README.md:17]: Wrap in triple backticks

File: docker-aoi-compose/docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+                tty: true
+        db:
+                image: mysql
+                networks:
+                        - aoinet
+                command: --default-authentication-plugin=mysql_native_password
+                environment:
+                        MYSQL_ROOT_PASSWORD: example
+networks:
+        aoinet:
```
Comment (kchoadley) [inline docker-aoi-compose/docker-compose.yaml:22]: I assume declaring the network name is all that is necessary?

File: rest-server/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+find_package(Threads REQUIRED)
+find_package(OpenSSL 1.0.1 REQUIRED)
+set(OPENSSL_VERSION "1.0.1")
+
+message("-- configuring gcc options")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++17 -DBOOST_LOG_DYN_LINK")
+
+# source files
+add_executable(${PROJECT_NAME}  
+                                ./src/controller/user_controller.cpp
```
Comment (kchoadley) [inline rest-server/CMakeLists.txt]: The executables should be in alphabetical order

File: rest-server/src/controller/example_controller.cpp [hunk 8]
```diff
@@ -0,0 +1,44 @@
+// Copyright 2018   Vaniya Agrawal, Ross Arcemont, Kristofer Hoadley,
+//                  Shawn Hulce, Michael McCulley
+#include <cpprest/uri_builder.h>
+#include <cpprest/base_uri.h>
+#include "example_controller.hpp"
+
+namespace aoi_rest {
+void ExampleController::InitHandlers() {
+    listener__.support(methods::GET,
```
Comment (kchoadley) [inline rest-server/src/controller/example_controller.cpp]: One line

File: rest-server/src/rest_server.cpp [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+#include "data_access_object.hpp"
+using aoi_rest::DataAccessInterface;
+using aoi_rest::DataAccessObject;
+using aoi_rest::AnotherController;
+using aoi_rest::ExampleController;
+using aoi_rest::UserController;
+
+int main(int argc, const char * argv[]) {
+    std::cout << "Starting aoi rest server" << std::endl;
+    DataAccessInterface* user_dao = &DataAccessObject::Instance();
```
Comment (kchoadley) [inline rest-server/src/rest_server.cpp:18]: Does this need to use a pointer?

File: .travis.yml (hunk 1 above)
Comment (SHulce) [reply .travis.yml:6]: Done

File: docker-aoi-apache/Dockerfile (hunk 2 above)
Comment (SHulce) [reply docker-aoi-apache/Dockerfile]: Added.

File: docker-aoi-apache/README.md (hunk 3 above)
Comment (SHulce) [reply docker-aoi-apache/README.md]: Added around each section of instructions

File: docker-aoi-apache/README.md (hunk 4 above)
Comment (SHulce) [reply docker-aoi-apache/README.md:11]: Prototype is only tag available for this image. Will leave for now

File: docker-aoi-apache/README.md (hunk 5 above)
Comment (SHulce) [reply docker-aoi-apache/README.md:17]: Done

File: docker-aoi-compose/docker-compose.yaml (hunk 6 above)
Comment (SHulce) [reply docker-aoi-compose/docker-compose.yaml:22]: Yes, the network name declaration enables the custom network with default parameters, but it must be explicitly declared.

File: rest-server/CMakeLists.txt (hunk 7 above)
Comment (SHulce) [reply rest-server/CMakeLists.txt]: 👍

File: rest-server/src/rest_server.cpp (hunk 9 above)
Comment (SHulce) [reply rest-server/src/rest_server.cpp:18]: I believe it does, since it is a single object, any access to it would be pointers no?

**Review by SHulce — CHANGES_REQUESTED** [review]
Modifications being done in branch issue-80. Once committed to dev branch merge can be completed.

**Review by SHulce — APPROVED** [review]

**Pull request merged**